test/unit/decoded_view.cpp

```diff
@@ -0,0 +1,508 @@
+            BOOST_TEST(s != str2);
+            BOOST_TEST_NOT(s != str0);
+            BOOST_TEST(s != bstr2);
+            BOOST_TEST_NOT(s != bstr0);
+
+            BOOST_TEST(s < s2);
+            BOOST_TEST_NOT(s < s0);
+            BOOST_TEST(s < str2);
+            BOOST_TEST_NOT(s < str0);
+            BOOST_TEST(s < bstr2);
+            BOOST_TEST_NOT(s < bstr0);
+
+            BOOST_TEST(s <= s2);
+            BOOST_TEST(s <= s0);
+            BOOST_TEST(s <= str2);
+            BOOST_TEST(s <= str0);
+            BOOST_TEST(s <= bstr2);
+            BOOST_TEST(s <= bstr0);
+
+            BOOST_TEST(s > s1);
+            BOOST_TEST_NOT(s > s0);
+            BOOST_TEST(s > str1);
+            BOOST_TEST_NOT(s > str0);
+            BOOST_TEST(s > bstr1);
+            BOOST_TEST_NOT(s > bstr0);
+
+            BOOST_TEST(s >= s1);
+            BOOST_TEST(s >= s0);
+            BOOST_TEST(s >= str1);
+            BOOST_TEST(s >= str0);
+            BOOST_TEST(s >= bstr1);
+            BOOST_TEST(s >= bstr0);
+
+        }
+    }
+
+    void
+    testConversion()
+    {
+        // to_string()
+        {
+            decoded_view s(str);
+            std::string o = "to be recycled";
+            BOOST_TEST_EQ(s.to_string(o), dec_str);
+        }
+
+        // to_string()
+        {
+            decoded_view s(str);
+            BOOST_TEST_EQ(s.to_string(), dec_str);
+        }
+
+        // explicit operator basic_string()
+        {
+            decoded_view s(str);
+            std::string o(s);
+            BOOST_TEST_EQ(o, dec_str);
+        }
+
+        // pass it to a function taking a string_view
+        {
+            auto fn = [this](string_view sv)
+            {
+                BOOST_TEST(sv == dec_str);
+            };
+            decoded_view s(str);
+            fn(s.to_string());
+        }
+
+        // pass it to a function taking a char*
+        {
+            auto fn = [this](char const* sv)
+            {
+                BOOST_TEST(sv == dec_str);
+            };
+            decoded_view s(str);
+            fn(s.to_string().c_str());
+        }
+    }
+
+    void
+    testAppend()
+    {
+        // to_string()
+        {
+            decoded_view s(str);
+            std::string o = "init ";
+            s.append_to(o);
+            std::string exp = std::string("init ");
+            exp += dec_str;
+            BOOST_TEST_EQ(o, exp);
+        }
+    }
+
+    void
+    testStream()
+    {
+        // operator<<
+        {
+            std::stringstream ss;
+            decoded_view      s(str);
+            ss << s;
+            BOOST_TEST_EQ(ss.str(), dec_str);
+        }
+    }
+
+    void
+    testPR127Cases()
+    {
+        {
+            std::stringstream ss;
+            urls::decoded_view ds("test+string");
+            // no warning about object slicing
+            ss << ds;
+        }
+
+        {
+            BOOST_STATIC_ASSERT(
+                sizeof(urls::decoded_view) ==
```

Comment:
eh.. this static assert is questionable

Reply:
Oh... this is something I replicated from PR127 to prove a point. It doesn't make any sense anymore.